codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -142,8 +144,33 @@ public int numActiveStreams() {
     }
 
     @Override
-    public Set<Http2Stream> activeStreams() {
-        return Collections.unmodifiableSet(activeStreams);
+    public Http2Stream forEachActiveStream(ActiveStreamVisitor visitor) throws Http2Exception {
+        ++activeStreamsIterationCount;
+        Http2Stream resultStream = null;
+        try {
+            for (Http2Stream stream : activeStreams) {
+                if (!visitor.visit(stream)) {
+                    resultStream = stream;
+                    break;
+                }
+            }
+            return resultStream;
+        } finally {
+            --activeStreamsIterationCount;
+            if (canModifyActiveStreams()) {
+                for (;;) {
+                    ActiveStreamsEvent event = activeStreamsPendingEvents.poll();
+                    if (event == null) {
+                        break;
+                    }
+                    event.process();
```

Comment:
This will be part of the "connection closes" if exception happens as part of https://github.com/netty/netty/issues/3501#issuecomment-88213605.